scene/2d/skeleton_2d.cpp

```diff
@@ -30,9 +30,20 @@
 
 #include "skeleton_2d.h"
 
+void Bone2D::_order_changed_in_parent() {
+	if (skeleton) {
+		skeleton->_make_transform_dirty();
+	}
+}
+
 void Bone2D::_notification(int p_what) {
 	if (p_what == NOTIFICATION_ENTER_TREE) {
 		Node *parent = get_parent();
+
+		if (parent) {
+			parent->connect("child_order_changed", this, "_order_changed_in_parent");
+		}
```

Comment:
This could potentially directly connect the skeleton as in master with a ref counted connection, however doing it via the `Bone2D` as before the PR is less error prone.

Could possibly do a direct skeleton connection in follow up PR.